tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -78,6 +79,26 @@ class resource_type(ArmResourceManager.resource_type):
         resource_type = 'Microsoft.DocumentDB/databaseAccounts'
 
 
+@CosmosDB.filter_registry.register('firewall-rules')
+class CosmosDBFirewallRulesFilter(FirewallRulesFilter):
+
+    def __init__(self, data, manager=None):
+        super(CosmosDBFirewallRulesFilter, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.cosmosdb')
```

Comment:
logs are singletons, its better to instantiate at module or class scopes rather than per instance.

Reply:
Sure, yeah I didn't add this I just moved things around in this file. I'll figure out why it was added this way and update accordingly!

Reply:
We have been doing them that way for subclassed filters/actions, but just setting `log` in the module would probably be good.    This isn't part of your change though so don't go update them all before your commit (you'd have to edit every function that I have in another pending PR)

Reply:
Filed #4644 to go clean these up, there are lots of them.